frontend-react/src/contexts/SessionContext.tsx

```diff
@@ -43,8 +44,8 @@ const SessionProvider = ({
         state: { activeMembership, initialized },
         dispatch,
     } = useOktaMemberships(authState);
-    /* This logic is a for when admins have other orgs present on their Okta claims
-     * that interfere with the activeMembership.memberType "soft" check */
+    /* TODO: Remove - logic can be easily checked by accessing the session context
```

Comment:
Hmmm I feel like it's still a nice convenience value in Context, no?  We could just check `memberType` wherever we need it, but we'd have to add a comparison against `MemberType.PRIME_ADMIN` every time.  My feeling is that it's easier just to have this boolean set in one place even if it's technically redundant 🤷

Reply:
It's just weird to have it only for admins, and _not_ have an equivalent for other member types. I was thinking it made more sense to condense the conditionals on member type to instances like `AuthElement` or that `requiresServices` HOF idea (related to gating admins from pages requiring mocked state)

Reply:
In that case, I'd argue for an `is*` predicate for each of our member types 🤷 

```js
isAdmin
isSender
isReceiver
```

I don't think HOC/HOF will have enough granularity to handle alllllll cases down the component hierarchy -- especially in the cases of individually swapped-in components or hooks.  These convenience predicates could also be used in HOCs/HOFs, but they'll give us more flexibility.

Reply:
Oh, I think what you're suggesting here goes well with what I'm suggesting! I'm not saying we need a catch-all HOF, I'm just saying using the checks in a consistent way (i.e. not having _only_ an `isAdmin` while relying on `membeType === xyz` for the other types) is preferred.

I'll add in the other two types, and I can make a separate PR implementing those like I'll do for implementing services :)

Reply:
Actually, to avoid scope creep, I'll alter the TODO here and make a ticket to link to instead